### ISSUE TITLE
✨ Be more selective with parts of screen hidden by loading spinner

### ISSUE THREAD

**Issue by tristan-orourke** [GCTC-NTGC/gc-digital-talent#6650]
## ✨ Feature

Right now, when any component is `pending`, a loading spinner covers the entire screen. But often (usually?) a portion of the UI could be rendered earlier, and progressively smaller blocks of the UI would need to be blocked by a loading spinner. I wonder if this would help the site feel snappier and more responsive.

I tested this by defaulting the `inline` prop to true instead of false in the `<Pending>` component, and it looks a little janky but is already close to what I'm thinking about. The biggest obvious improvement is that the user doesn't lose access to nav menus while pages are loading.

## 🕵️ Details

Add any additional details that could assist with the development of the new feature.

## 🎨 Design File

Include a link to the design file (if it exists).

## 📸 Screenshot

Add a screenshot of the new feature (if possible).

## 🙋‍♀️ Proposed Implementation

Mostly, this would just require some tweaks to the Loading and/or Pending components. 

We'd also want to audit the UI to see how this changes things. For example, I noticed while testing with  `inline = true` that the whole layout (including the stepper) is replaced by a spinner when you save changes in any step. The stepper comes back earlier, but blinking away like that might be more disruptive than the current spinner-over-entire-page. Speaking of which, for forms like this, should we work on just showing the spinner in the button while things are saving/refreshing, instead of blocking out the entire page?

## 🌎 Localization

(optional) Provide any new copy along with translations available.

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the feature was properly implemented.
- [ ] Criteria 1
- [ ] Criteria 2

**Comment by tristan-orourke** [comment]
Hey @Jerryescandon and @substrae , here's a little proposal for you to think about.

**Comment by substrae** [comment]
I'm all for inline loading indicators. The original button component had a design for this, but we could adapt it for our current buttons really easily:

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/6960386/68ac40af-e321-4ddc-ae76-764f09f08dcf)

The only thing that sticks out is the example you provided where the stepper disappears in place of the loader - that's not a good experience, so situations like this will need custom treatment.

**Comment by mnigh** [comment]
@tristan-orourke is this issue still relevant? if so, what's the acceptance criteria?

**Comment by tristan-orourke** [comment]
I believe it's still relevant, but it needs more design/decision work

**Comment by mnigh** [comment]
> I believe it's still relevant, but it needs more design/decision work

@tristan-orourke Maybe worth moving to https://github.com/GCTC-NTGC/design-gc-digital-talent?

**Comment by tristan-orourke** [comment]
This probably needs some cooperative brainstorming between a dev and a designer if its to get anywhere.

**Comment by gobyrne** [comment]
Yeah, this is valuable, but not something we will work on soon. Moving to pre-dev backlog.
